tests/operations/test_operations.py

```diff
@@ -0,0 +1,34 @@
+from myia import myia
+
+
+@myia
+def run_lsfhit(a, b):
+    return a << b
+
+
+@myia
+def run_rshift(a, b):
+    return a >> b
+
+
+@myia
+def run_bit_and(a, b):
+    return a & b
+
+
+@myia
+def run_bit_or(a, b):
+    return a | b
+
+
+@myia
+def run_bit_xor(a, b):
+    return a ^ b
+
+
+def test_bitwise_operations():
+    assert run_lsfhit(3, 2) == 12
+    assert run_rshift(12, 2) == 3
+    assert run_bit_and(5, 7) == 5
+    assert run_bit_or(5, 2) == 7
+    assert run_bit_xor(10, 8) == 2
```

Comment:
Each operation should be tested separately. You should use the `@mt` decorator defined in `tests/multitest.py`, look in `tests/test_algos.py` for examples. Something like this, basically:

```python
@mt(
    infer(i64, i64, result=i64),  # test type inference
    run(10, 8, result=2),         # test implementation and result
    # other tests go here
)
def test_bit_xor(a, b):
    return a ^ b
```

This will also test all backends, although there is also `run_debug` and `run_no_relay`. I believe you don't have Relay support, but I think it should be as easy to add as for Pytorch, assuming Relay supports these operations at all.

You can omit the result in `run()`, in that case it'll compare to the Python result.

Reply:
Relay supports right and left shift but doesn't have the others.